test_collections/sdk_tests/support/python_testing/models/python_test_parser.py

```diff
@@ -0,0 +1,119 @@
+#
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import ast
+from pathlib import Path
+from typing import List
+
+from test_collections.sdk_tests.support.models.matter_test_models import (
+    MatterTestStep,
+    MatterTestType,
+)
+
+from .python_test_models import PythonTest
+
+ARG_STEP_DESCRIPTION_INDEX = 1
+KEYWORD_IS_COMISSIONING_INDEX = 0
+BODY_INDEX = 0
+
+
+class PythonParserException(Exception):
+    """Raised when an error occurs during the parser of python file."""
+
+
+def parse_python_test(path: Path) -> PythonTest:
+    """Parse a single Python test file into PythonTest model.
+
+    This will also annotate parsed python test with it's path and test type.
+    """
+
+    return __parse_test_case_from_file(path)
```

Comment:
Why not make `__parse_test_case_from_file` public and use it directly? Or maybe keep this function name and use `__parse_test_case_from_file`'s body